src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySource.java

```diff
@@ -136,11 +146,19 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     configuration.setBoolean("fs.gs.impl.disable.cache", true);
     configuration.setBoolean("fs.gs.metadata.cache.enable", false);
 
+    String query = generateQuery();
```

Comment:
This logic should be moved into the InputFormat. The InputFormat should support additional partition start and end settings, with the query generation logic and temp table logic entirely in the InputFormat. The source should only be responsible for setting the partition start and end settings.

Reply:
Done.